lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -205,11 +208,26 @@ public Builder setDimensions(DimensionList dimensions) {
      * (Optional) Set the timestamp for the exported metric line. In most cases, {@link
      * Builder#setCurrentTime()} should be suitable.
      *
+     * <p>If the timestamp is from before the year 2000 or from after the year 3000 * (e.g., when
+     * the wrong unit was used when creating the {@link Instant}), * the timestamp will be discarded
+     * and no value will be set.
+     *
      * @param timestamp an {@link Instant} object describing the time at which the {@link Metric}
      *     was created.
      * @return this
      */
     public Builder setTimestamp(Instant timestamp) {
+      int year = timestamp.atZone(ZoneOffset.UTC).getYear();
+      if (year < 2000 || year > 3000) {
+        logger.warning(
```

Comment:
This will be quite a verbose log if you are receiving many metrics with the incorrect timestamp. Perhaps we can rate limit it?

Reply:
Added a rate limiter that prints only 1 in 1000 messages.